### PR TITLE
forever.config Configstore init and usage fixes

### PR DESCRIPTION
**Pull request by akatona84**
Fixes: TypeError: forever.config.saveSync is not a function

### CODE REVIEW

**Comment by akatona84** [comment]
fixes: #1114
@kibertoad, @thenengah could you review my change, please?

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ forever start app.js
     columns add <col>   Adds the specified column to the output in `forever list`. Supported columns: 'uid', 'command', 'script', 'forever', 'pid', 'id', 'logfile', 'uptime'
     columns rm <col>    Removed the specified column from the output in `forever list`
     columns set <cols>  Set all columns for the output in `forever list`
+    columns reset       Resets all columns to defaults for the output in `forever list`
```
[the remark this answers is not in the document]
Comment (akatona84) [reply README.md:59]: I just noticed that it wasn't part of the readme but the usage prints it, so I synched it.

**Comment by akatona84** [comment]
@bmeck , @indexzero , @jcrugzz could u take a look too? 
Sorry for pushing u guys, but we really need a fix and a new (latest) version.
Many thanks!

**Comment by kibertoad** [comment]
I'll review today, please let me catch my breath a bit :)

**Comment by akatona84** [comment]
> I'll review today, please let me catch my breath a bit :)

Sorrrry 😅

**Comment by kibertoad** [comment]
@akatona84 Would it be possible to also add test for the main fix, the one related to the configStore error?

File: lib/util/config-utils.js [hunk 2]
```diff
@@ -4,7 +4,9 @@ var Configstore = require('configstore');
 
 function initConfigFile(foreverRoot) {
   try {
-    return new Configstore(JSON.parse(fs.readFileSync(path.join(foreverRoot, 'config.json'))));
+    return new Configstore('config', undefined, {
```
Comment (kibertoad) [inline lib/util/config-utils.js:7]: What's the idea behind this change?

File: lib/forever/cli.js [hunk 3]
```diff
@@ -554,7 +554,6 @@ app.cmd('columns add :name', cli.addColumn = function (name) {
     columns.push(name);
 
     forever.config.set('columns', columns);
-    forever.config.saveSync();
```
Comment (kibertoad) [inline lib/forever/cli.js:557]: Why are all of save calls removed without any replacement?

File: lib/forever/cli.js [hunk 4]
```diff
@@ -444,8 +444,8 @@ app.cmd('list', cli.list = function () {
 // Lists all of the configuration in `~/.forever/config.json`.
 //
 app.cmd('config', cli.config = function () {
-  var keys = Object.keys(forever.config.store),
-      conf = cliff.inspect(forever.config.store);
+  var keys = Object.keys(forever.config.all),
+      conf = cliff.inspect(forever.config.all);
```
Comment (akatona84) [inline lib/forever/cli.js:448]: Configstore doesn't have store attribute, all() returns the whole

File: lib/forever/cli.js (hunk 3 above)
Comment (akatona84) [reply lib/forever/cli.js:557]: Configstore persists the added configs right away, it doesn't require explicit save method call to save it to the file.

File: lib/util/config-utils.js (hunk 2 above)
Comment (akatona84) [reply lib/util/config-utils.js:7]: before this nconf was used, like this:
```
function initConfigFile(foreverRoot) {
  return new nconf.File({file: path.join(foreverRoot, 'config.json')});
}
```
whatever changes you made in the configuration, it was saved eventually in this config.json file.
The first param should be an `id` and then the `defaults` and then `options`. It passed the parsed json to the first param before, which was used for creating the config file appending a ".json" to it so it became '[object Object].json' in `~/.config/configstore` dir.
No backward compatibility, default configs was saved to this oddly named json and config.json in .forever dir was completely ignored.
I made it backward compatible and to use the original config file location.

File: test/mocha/cli/cli.spec.js [hunk 5]
```diff
@@ -3,6 +3,33 @@ const { delayPromise } = require('../../helpers/utils');
 const { execCommand } = require('cli-testlab');
 
 describe('cli', () => {
+  describe('columns', function () {
+    this.timeout(5000);
+    it('manages columns successfully', function () {
+      return execCommand(`node bin/forever columns reset`,
```
Comment (akatona84) [inline test/mocha/cli/cli.spec.js:9]: this is testing whether `forever columns reset` works, it was throwing `TypeError: forever.config.saveSync is not a function` originally

File: test/mocha/cli/cli.spec.js [hunk 6]
```diff
@@ -3,6 +3,33 @@ const { delayPromise } = require('../../helpers/utils');
...
+            return execCommand(`node bin/forever columns rm uptime`,
+                {
+                  expectedOutput: ['Removing column:', 'uptime']
+                });
+          })
+          .then(function () {
+            return execCommand(`node bin/forever columns add uptime`,
+                {
+                  expectedOutput: ['Adding column:', 'uptime']
+                });
```
Comment (akatona84) [inline test/mocha/cli/cli.spec.js:23]: these two just testing whether these commands are working too

File: test/mocha/cli/cli.spec.js [hunk 7]
```diff
@@ -3,6 +3,33 @@ const { delayPromise } = require('../../helpers/utils');
...
+          })
+          .then(function () {
+            return execCommand(`node bin/forever columns add uptime`,
```
Comment (akatona84) [inline test/mocha/cli/cli.spec.js:26]: also checking config persistence and a bit of functionality

**Comment by akatona84** [comment]
> @akatona84 Would it be possible to also add test for the main fix, the one related to the configStore error?

The tests I added are using the Configstore and previously those commands are failing with 4.0.0.
(TBH the migration from nconf wasn't a too sophisticated change, this PR is basically finishing it. The errors were happening  because forever code was still using some calls which were still nconf specific.)

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
@akatona84 Thanks a lot for contributing the fix!

**Comment by akatona84** [comment]
@kibertoad, Thank you for your help and your quick actions! 🍻

**Pull request merged**